source/auxiliaries.ts

```diff
@@ -275,7 +275,7 @@ namespace auxiliaries {
      * Queries window.location.search.
      */
     export function GETsearch(): string {
-        return document.location.search;
+        return document.location!.search;
```

Comment:
With `window.location` you don't need an exclamation mark ([should be the same](https://stackoverflow.com/a/2430948/2858790)).
Same below.